design/unnumbered-bgp.md

```diff
@@ -0,0 +1,436 @@
+
+## Proposal
+
+### User Stories
+
+There is direct link interface between the node (FRR process acting as BGP peer) and the
+ToR (acting as an BGP router).
+
+
+#### Story 1 - Unnumbered {i,E}BGP on primary node interface
+
+```mermaid
+graph TD
+    subgraph Rack0
+        direction LR
+        subgraph ToR0
+            direction TB
+            A1[eth0] -.- A{Routing}
+            A -.- A3[eth00]
+            A -.- A4[eth01]
+
+            BGP(BGP)
+
+        end
+
+        %% Top Right Subgraph
+        subgraph Node1
+            direction TB
+            B1 --- B2(cni)
+            B2 --- B3
+            subgraph Pod1[pod]
+              B3[eth0]
+            end
+        end
+
+        %% Bottom Right Subgraph
+        subgraph Node0
+            direction TB
+            C1 --- C2(cni)
+            C2 --- C3
+            subgraph Pod2[pod]
+              C3[eth0]
+            end
+        end
+
+        %% Connections between subgraphs
+        A3 -- point-to-point ------ B1[eth0]
+        A4 -- point-to-point ------ C1[eth0]
+    end
+
+    %% Define styles for subgraphs and nodes
+    classDef node fill:#f0f0f0,stroke:#000000,stroke-width:1px;
+    class Node0 node;
+    class Node1 node;
+```
+
+#### Story 2 - Unnumbered {i,E}BGP on secondary node interface
+
+```mermaid
+graph TD
+    subgraph Rack0
+        direction LR
+        style Rack0 fill:#ffffff,stroke:#000000,stroke-width:2px
+
+        subgraph ToR0
+            style ToR0 fill:#ffffff,stroke:#000000,stroke-width:2px
+            direction TB
+            subgraph Default
+              direction LR
+              A1[net0] -.- A{Routing}
+              A -.- A3[eth00]
+              A -.- A4[eth01]
+            end
+            subgraph RED VRF*
+              direction LR
+              AA1[net00] -.- AA{Routing}
+              AA -.- AA3[eth00.vlanRED]
+              AA -.- AA4[eth01.vlanRED]
+              BGP-T0(BGP configured on RED)
+            end
+
+        end
+
+        subgraph Node0
+            direction TB
+            B1 --- B2(cni)
+            B2 --- B3
+            BB1
+            subgraph Pod1[pod]
+              B3[eth0]
+            end
+            BGP-N0(BGP process)
+        end
+
+        subgraph Node1
+            direction TB
+            C1 --- C2(cni)
+            C2 --- C3
+            CC1
+            subgraph Pod2[pod]
+              C3[eth0]
+            end
+            BGP-N1(BGP process)
+        end
+
+        %% Connections between subgraphs
+        A3 ------- B1[eth0]
+        AA3 -- BGP Peering ----- BB1[net0]
+        A4 ------- C1[eth0]
+        AA4 -- BGP Peering ----- CC1[net0]
+    end
+
+    %% Define styles for subgraphs and nodes
+    classDef node fill:#f0f0f0,stroke:#000000,stroke-width:1px;
+    class Node0 node;
+    class Node1 node;
+```
+
+## Design details
+
+The target `show running-config` in the FRR process running in the k8s node
```

Comment:
can you make this more alike the config we have today for the speakers? also, can you add a few words about the need for peergroups (iirc it's because the usual directives do not work with the interface's name)

Reply:
I will go deeper on what we have today in the speaker, that part if not very clear.

About PeerGroups:
1. Peergroups is the only way I managed to make it working, and all reference/examples did make use of them. So I observe we need them but I did not find doc.
2. I think the case of having two ToRs switch will be a needed, so I am thinking that we can have PG with two interfaces assigned.

Reply:
> Peergroups is the only way I managed to make it working, and all reference/examples did make use of them. So I observe we need them but I did not find doc.

right, this is what I meant by "the usual directives do not work with the interface's name" :) let's add this to this section so it's clear why we suddenly need peer groups.

> I think the case of having two ToRs switch will be a needed, so I am thinking that we can have PG with two interfaces assigned.

I see, but I guess this could be achieved by just having 2 different peers?

Reply:
ok, I will go through to see how two neighbors would look like, and check again if Peer Group can be avoided